### PR TITLE
remove concept of alias domains

### PR DESCRIPTION
**Pull request by tkellen**
Instead, provide an array of domains and explcitly select one of
them as the canonical based on the environment. This prevents
issues with nginx config being generated incorrectly in staging.

### CODE REVIEW

**Comment by tkellen** [comment]
cc @cowboy for review on this minor adjustment. Tested and confirmed functioning on production/staging (now testing on vagrant).

**Comment by cowboy** [comment]
@tkellen I don't really like the idea of storing the domain name in the vars file, I think that's best done in individual inventory files. That way, any arbitrary inventory file can be created for testing. Actually, what if you stored the array of domain names in each inventory file?

**Comment by tkellen** [comment]
@cowboy Looked at doing that too. The inventory files would wind up being defined thusly:
```
ajl.bocoup.org env=production domains="['ajl.bocoup.org','fightbias.io','fightbias.org']"
```

I don't feel super strongly about one approach vs the other. I thought the syntax in the inventory file looked strange, but I do see the flexibility it would afford.

Curious to hear what you think @gnarf.

**Comment by gnarf** [comment]
> I don't feel super strongly about one approach vs the other. I thought the syntax in the inventory file looked strange, but I do see the flexibility it could afford.

👍  this seems fine too (defnining domains in inventory) -- I think it just looks kinda weird for `domains="['ajl-staging.bocoup.org']"`

**Comment by tkellen** [comment]
coo, updating now.

File: ansible/templates/nginx.conf.j2 [hunk 1]
```diff
@@ -59,7 +59,7 @@ server {
 server {
   server_name {{domain}};
   listen 80;
-  return 301 https://{{fqdn}}$request_uri;
+  return 301 https://{{canonical_domain}}$request_uri;
```
Comment (cowboy) [inline ansible/templates/nginx.conf.j2]: Should this be `domain`? If it's `canonical_domain`, we can probably get rid of all the non-canonical domain `ssl` server blocks in production. I guess I missed this in my initial PR.

File: ansible/vars/main.yml [hunk 2]
```diff
@@ -60,11 +72,13 @@ basic_auth: "{{env != 'development'}}"
...
-  - fightbias.org
-  - fightbias.io
-  - ajl.ai
+# These variables control which domains nginx listens for.
+# If a domain list is provided in an inventory file, nginx will
+# use the first entry as a the canonical and redirect all others
+# to it. If no domain listing is provided it is assumed that the
+# provided hostname in the inventory file is the only domain.
+domains: "{{domain_list | default([inventory_hostname])}}"
+canonical_domain: "{{domains.0 | inventory_hostname}}"
```
Comment (cowboy) [inline ansible/vars/main.yml]: Can this just be `"{{domains.0}}"`?

File: ansible/templates/nginx.conf.j2 (hunk 1 above)
Comment (tkellen) [inline ansible/templates/nginx.conf.j2]: This works as is, but it *would* be possible to have a single port 80 catchall that doesn't even specify a server name and redirects anything that hits it to the canonical.

**Comment by tkellen** [comment]
Okay @cowboy, all of your feedback has been incorporated. Mind having another look?

**Comment by tkellen** [comment]
Just noting that I asked mr. sender to do this over the phone on friday in a state of confusion about an issue he was having with deployment. I'd still like your review @cowboy!

**Comment by cowboy** [comment]
@tkellen was the dehydrated process actually tested for multiple domains with this new setup? Because I don't see anything in nginx that gets well-known working with non-canonical domains on port 80 (which is how I had it set up before, at least).

**Comment by cowboy** [comment]
(If you _do_ test this, please be sure to use the staging letsencrypt urls)

**Pull request merged**